File: problems/0442 - Find All Duplicates in an Array/Go/Solution.go
func findDuplicates(nums []int) []int {
    var res = make([]int, 0)
    if len(nums) == 1 {
        return res
    }
    
    for _, v := range nums {
		v = int(math.Abs(float64(v)))
		if nums[v-1] > 0 {
			nums[v-1] = nums[v-1] * -1
		} else {
			res = append(res, v)
		}
	}
    
    return res
}

File: problems/0442 - Find All Duplicates in an Array/PHP/Solution.php
<?php

class Solution {

    /**
     * @param Integer[] $nums
     * @return Integer[]
     */
    function findDuplicates($nums) {
        $res = [];
        if (count($nums) === 1) {
            return $res;
        }

        foreach ($nums as $num) {
            $num = (int)$num;
            if ($nums[$num-1] > 0) {
                $nums[$num-1] = $nums[$num-1]*-1;
            } else {
                $res[] = $num;
            }
        }

        return $res;
    }
}